### PR TITLE
kill most todos

### PR DESCRIPTION
**Pull request by stephanenicolas**

### CODE REVIEW

File: toothpick-compiler/src/main/java/toothpick/compiler/factory/FactoryProcessor.java [hunk 1]
```diff
@@ -113,9 +111,26 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
...
   private void findAndParseTargets(RoundEnvironment roundEnv) {
     createFactoriesForClassesWithInjectAnnotatedConstructors(roundEnv);
+    createFactoriesForClassesAnnotatedProvidesSingleton(roundEnv);
+    createFactoriesForClassesAnnotatedSingleton(roundEnv);
+
     createOptimisticFactories(roundEnv);
   }
 
+  private void createFactoriesForClassesAnnotatedProvidesSingleton(RoundEnvironment roundEnv) {
+    for (Element singletonAnnotatedElement : roundEnv.getElementsAnnotatedWith(Singleton.class)) {
```
Comment (dlemures) [inline toothpick-compiler/src/main/java/toothpick/compiler/factory/FactoryProcessor.java]: providesSingletonAnnotatedElement

File: toothpick-compiler/src/main/java/toothpick/compiler/factory/FactoryProcessor.java [hunk 2]
```diff
@@ -113,9 +111,26 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
...
+      TypeElement singletonAnnotatedTypeElement = (TypeElement) singletonAnnotatedElement;
+      processClassContainingInjectAnnotatedMember(singletonAnnotatedTypeElement, mapTypeElementToConstructorInjectionTarget);
+    }
+  }
+
+  private void createFactoriesForClassesAnnotatedSingleton(RoundEnvironment roundEnv) {
```
Comment (dlemures) [inline toothpick-compiler/src/main/java/toothpick/compiler/factory/FactoryProcessor.java]: opposite here

File: toothpick-compiler/src/main/java/toothpick/compiler/factory/FactoryProcessor.java [hunk 3]
```diff
@@ -267,25 +273,27 @@ private void processInjectAnnotatedParameters(ExecutableElement methodElement,
...
     Set<Modifier> parentModifiers = enclosingElement.getModifiers();
-    //TODO should not be a non static inner class neither
     if (!parentModifiers.contains(PUBLIC)) {
       error(element, "Class %s is private. @Inject constructors are not allowed in non public classes.", enclosingElement.getQualifiedName());
-      valid = false;
+      return false;
     }
 
-    return valid;
+    if (isNonStaticInnerClass(enclosingElement)) {
```
Comment (dlemures) [inline toothpick-compiler/src/main/java/toothpick/compiler/factory/FactoryProcessor.java]: shouldn't all errors be displayed?

**Comment by dlemures** [comment]
small comments and RTM

File: toothpick-compiler/src/main/java/toothpick/compiler/factory/FactoryProcessor.java (hunk 3 above)
Comment (stephanenicolas) [inline toothpick-compiler/src/main/java/toothpick/compiler/factory/FactoryProcessor.java]: they are in isNonStaticInnerClass, it was easier at this point.

**Comment by stephanenicolas** [comment]
Good catches.

**Pull request closed without merging**